File: Data Structure/String/Medium/LET_no609_Find Duplicate File in System.cpp
class Solution {
public:
    vector<vector<string>> findDuplicate(vector<string>& paths) {
        unordered_map<string, vector<string>> records;
        vector<vector<string>> res;
        
        for (auto path : paths) {
            stringstream ss(path);
            string root;
            string s;
            getline(ss, root, ' ');
            while (getline(ss, s, ' ')) {
                string fileName = s.substr(0, s.find('('));
                string content = s.substr(s.find('(') + 1, s.find(')') - s.find('(') - 1);
                records[content].push_back(root + "/" + fileName);
            }
        }
        
        for (auto path : records) {
            if (path.second.size() >= 2)
                res.push_back(path.second);
        }
        
        return res;
    }
};

File: Data Structure/String/Medium/LET_no609_Find Duplicate File in System.py
class Solution:
    def findDuplicate(self, paths: List[str]) -> List[List[str]]:
        memo = collections.defaultdict(list)
        for p in paths:
            p = p.split()
            folder = p[0]
            for i in range(1, len(p)):
                file, content = p[i].split('(')
                memo[content[:-1]].append(folder + '/' + file)
                
        return [values for values in memo.values() if len(values) > 1]